File: src/main/java/rocks/sakira/flamingo/register/Effects.java
package rocks.sakira.flamingo.register;

import net.minecraft.potion.Effect;
import net.minecraft.potion.EffectType;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import rocks.sakira.flamingo.Flamingo;
import rocks.sakira.flamingo.effects.ShrimpEffect;

public class Effects {
    public static final DeferredRegister<Effect> REGISTER = DeferredRegister.create(ForgeRegistries.POTIONS, Flamingo.MOD_ID);

    public static final RegistryObject<Effect> SHRIMP_EFFECT = REGISTER.register(
            "shrimp",

            () -> new ShrimpEffect(EffectType.HARMFUL, 0xFF96AB)
    );
}

File: src/main/kotlin/rocks/sakira/flamingo/register/Effects.kt
package rocks.sakira.flamingo.register

import net.minecraft.entity.effect.StatusEffect
import net.minecraft.entity.effect.StatusEffectType
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry
import rocks.sakira.flamingo.MOD_ID
import rocks.sakira.flamingo.effects.ShrimpEffect

object Effects {
    val SHRIMP_EFFECT = ShrimpEffect(StatusEffectType.HARMFUL, 0xFF96AB)

    fun register() {
        Registry.register(Registry.STATUS_EFFECT, Identifier(MOD_ID, "shrimp"), SHRIMP_EFFECT)
    }
}
